Dockerfile

```diff
@@ -74,6 +74,7 @@ WORKDIR /opt/netbox/netbox
 # Must set permissions for '/opt/netbox/netbox/media' directory
 # to g+w so that pictures can be uploaded to netbox.
 RUN mkdir -p static /opt/unit/state/ /opt/unit/tmp/ \
+      && chown -R unit:unit /opt/unit/tmp \
```

Comment:
Are you sure this helps? Because unit is not run as user `unit`, but as whatever we tell it to in the `docker-compose.yml`, by default with the user id `101`.

Shouldn't rather [the next line](https://github.com/netbox-community/netbox-docker/pull/509/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R78) be changed to `chmod -R o+w media /opt/unit/tmp`, i.e. that _others_ (to which uid `101` IMO belongs) can write to the directories.

@tobiasge, wdyt? Can you recall, why we chose `g+w`?

Reply:
I see what's happening now. If the container is started as root, which it is by default, unitd is able to drop privileges, and it drops to using the group `unit`, which means that `g+w` on the directories owned by root doesn't work. If I execute the container as `unit` (ie., 101), it can't drop privileges, so it ends up retaining the `root` group and can write to those directories.

Is there any reason not to set `USER unit` in the Dockerfile?

Reply:
I raised github issue #506 to alert about this problem. We ran into the same issue as you when trying to make some big bulk update queries, that do not fit into Nginx memory buffer.

We run the container as root as well (the default), and had to use a customized Dockerfile to introduce the same change you proposed and its working just fine now.

```
chown unit:unit -R /opt/unit/tmp
```

Reply:
If you're running in K8s, set the `securityContext.runAsUser` to `101` like the docker-compose does instead, which is what I did as soon as I saw the first reply to this PR and realized that docker-compose wasn't running with the default user.

Reply:
> If the container is started as root, which it is by default, unitd is able to drop privileges, and it drops to using the group unit, which means that g+w on the directories owned by root doesn't work. If I execute the container as unit (ie., 101), it can't drop privileges, so it ends up retaining the root group and can write to those directories.

Wouldn't the actual fix then be to tell `unit` to become the user `101` instead of the user `unit`? Either on [the NetBox application](https://unit.nginx.org/configuration/#applications) or [during startup](https://unit.nginx.org/installation/#installation-src-startup), of which I lean towards the latter.

Which would mean to change [the lines in `docker/launch-netbox.sh`](https://github.com/netbox-community/netbox-docker/blob/release/docker/launch-netbox.sh#L48-L54) to something like the following:

```bash
exec unitd \
          --no-daemon \
          --control unix:$UNIT_SOCKET \
          --pid /opt/unit/unit.pid \
          --log /dev/stdout \
          --state /opt/unit/state/ \
          --tmp /opt/unit/tmp/ \
          --user 101
```

or maybe

```bash
exec unitd \
          --no-daemon \
          --control unix:$UNIT_SOCKET \
          --pid /opt/unit/unit.pid \
          --log /dev/stdout \
          --state /opt/unit/state/ \
          --tmp /opt/unit/tmp/ \
          --group root
```

Reply:
If done that way, I would personally specify both `--user` and `--group`. That's a possible solution, although I'm not sure what happens when unitd tries to start with those explicit settings and finds that it can't drop privileges since it's not root... Right now it logs a warning and continues, but the settings aren't explicit.

But it seems safest to set the `USER` in the Dockerfile unless there's a reason not to. There are mechanisms in K8s to disallow running as root (actually, any UID, but 0 is surely the most commonly disallowed UID), so even with the `--user`/`--group` fix, someone might find that the container would refuse to start with default settings in such an environment. Somebody migrating from docker-compose to K8s might be surprised by this since docker-compose would be doing it for them.

Reference to the K8s policy I have in mind: https://kubernetes.io/docs/concepts/policy/pod-security-policy/#users-and-groups

> MustRunAsNonRoot - Requires that the pod be submitted with a non-zero runAsUser or have the USER directive defined (using a numeric UID) in the image. Pods which have specified neither runAsNonRoot nor runAsUser settings will be mutated to set runAsNonRoot=true, thus requiring a defined non-zero numeric USER directive in the container. No default provided. Setting allowPrivilegeEscalation=false is strongly recommended with this strategy.